### ISSUE TITLE
Istio Telemetry does not work

### ISSUE THREAD

**Issue by johgoe** [kyma-project/kyma#15875]
**Description**

**Expected result**

Tracing sample rate is adjusted by Telemetry CR.

**Actual result**

Using Telemetry from istio stops tracing at all.

**Steps to reproduce**

1 Add a telemetry CR like

```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: namespace-override
  namespace: myns
spec:
  tracing:
        userId:
          header:
            defaultValue: unknown
            name: x-user
      randomSamplingPercentage: 100
```

2) Check Jaeger,

**Troubleshooting**

I guess the problem is like https://github.com/istio/istio/issues/35750#issuecomment-1165988761

A possibile solution would be to name the extension provider and not use this provider for default configuration.

**Comment by johgoe** [comment]
@a-thaler Thank you for this informations.

**Comment by a-thaler** [comment]
The [configurable tracing](https://github.com/kyma-project/kyma/issues/11231) story will be released in a first version with Kyma 2.10. It will rely on the istio telemetry api by providing an extensionProvider, so that you can leverage the advanced configuration options.
For now, the provider will not be a default provider, so you will have to specify the provider explicitly. In a later step the istio operator will detect if there is tracing enabled and will configure it as a defaultProvider.

See also the upcoming new documentation: https://kyma-project.io/docs/kyma/main/01-overview/main-areas/telemetry/telemetry-03-traces/, feedback is welcome.

**Comment by chrkl** [comment]
With merging https://github.com/kyma-project/kyma/pull/16372, tracing can be configured with a Telemetry resource that references the `kyma-traces` extension provider. When using the tracing component as an in-cluster Jaeger backend, the `tracing.istio.samplingRate`  value can be used for configuration. Use [this](https://github.com/kyma-project/kyma/blob/main/resources/tracing/templates/kyma-additions/istio-telemetry.yaml) as an example to configure Istio tracing when using a custom backend.
